src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2435,7 +2435,8 @@ else if (!useTnir) {
 
         }
 
-        assert result.equals(Result.SUCCESS) == true;
+        boolean equalSuccess = result.equals(Result.SUCCESS);
```

Comment:
there is no usage of the variable. however, I think what sonar says is a good practice. In this case, the `equals()` method only check for equivalence and return a boolean value. if we think about another case: a method returns a boolean value and also modifies a class-level variable. In this case, if we use this method with assert, it is likely to cause issues if `-ea` is not enabled explicitly. Instead, we could always separate the method and the `assert` to make sure the method behaves correctly.  

I don't see any significant advantages in your suggestion over the original code.